setup.py

```diff
@@ -105,20 +104,24 @@ def run(self):
 if os.name == 'nt':
     scripts.append('scripts/mypy.bat')
 
-setup(name='mypy-lang',
-      version=version,
-      description=description,
-      long_description=long_description,
-      author='Jukka Lehtosalo',
-      author_email='jukka.lehtosalo@iki.fi',
-      url='http://www.mypy-lang.org/',
-      license='MIT License',
-      platforms=['POSIX'],
-      package_dir=package_dir,
-      py_modules=['typing'] if sys.version_info < (3, 5, 0) else [],
-      packages=['mypy'],
-      scripts=scripts,
-      data_files=data_files,
-      classifiers=classifiers,
-      cmdclass={'build_py': CustomPythonBuild},
-      )
+setup_kwargs = dict(
+    name='mypy-lang',
+    version=version,
+    description=description,
+    long_description=long_description,
+    author='Jukka Lehtosalo',
+    author_email='jukka.lehtosalo@iki.fi',
+    url='http://www.mypy-lang.org/',
+    license='MIT License',
+    platforms=['POSIX'],
+    package_dir=package_dir,
+    py_modules=['typing'] if sys.version_info < (3, 5, 0) else [],
+    packages=['mypy'],
+    scripts=scripts,
+    data_files=data_files,
+    classifiers=classifiers,
+    cmdclass={'build_py': CustomPythonBuild},
+)
+
+if __name__ == '__main__':
+    setup(**setup_kwargs)
```

Comment:
Oh, I don't like this at all. All this linkage between setup.py and extensions/setup.py is just extremely fragile.